File: leetcode2/2medium/유시연/2587.py
<<<<<<< HEAD
class Leetcode.이재훈.이재훈.Solution:
=======
class Leetcode.이재훈.Solution:
>>>>>>> ab7f46f73d8f18bcd38c5cf06ecabae64f53a424
    def maxScore(self, nums: List[int]) -> int:
        nums.sort(reverse=True)
        answer = 0
        score = 0
        for i in range(len(nums)):
            answer += nums[i]
            if answer >0:
                score += 1
            else:
                break
        
        return score

File: leetcode2/2medium/이준열/2587.cpp
class Solution {
    public:
        int maxScore(vector<int>& nums) {
            vector<long long> prefixSum(nums.size());
    
            sort(nums.begin(), nums.end(), std::greater<long long>());
            prefixSum[0] =nums[0];
    
            for ( long long i = 1; i < nums.size(); i++)
            {
                prefixSum[i] = nums[i] + prefixSum[i-1];
            }
    
            long long count = 0;
    
            for (long long prefix : prefixSum)
            {
                if (prefix > 0)
                    count++;
                else
                    break;
            }
    
            return count;
        }
    };